worlds/stardew_valley/logic/goal_logic.py

```diff
@@ -0,0 +1,176 @@
+import typing
+
+from .base_logic import BaseLogic, BaseLogicMixin
+from ..data.craftable_data import all_crafting_recipes_by_name
+from ..data.recipe_data import all_cooking_recipes_by_name
+from ..locations import LocationTags, locations_by_tag
+from ..mods.mod_data import ModNames
+from ..options import options
+from ..stardew_rule import StardewRule
+from ..strings.building_names import Building
+from ..strings.quest_names import Quest
+from ..strings.season_names import Season
+from ..strings.wallet_item_names import Wallet
+
+if typing.TYPE_CHECKING:
+    from .logic import StardewLogic
+else:
+    StardewLogic = object
+
+
+class GoalLogicMixin(BaseLogicMixin):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.goal = GoalLogic(*args, **kwargs)
+
+
+class GoalLogic(BaseLogic[StardewLogic]):
+
+    def can_complete_community_center(self) -> StardewRule:
+        return self.logic.bundle.can_complete_community_center
+
+    def can_finish_grandpa_evaluation(self) -> StardewRule:
+        # https://stardewvalleywiki.com/Grandpa
+        rules_worth_a_point = [
```

Comment:
I feel like half of these rules have, over time, been refactored and renamed, and are now wordy and clear enough that their related comment is completely redundant